test/test_io_common.py

```diff
@@ -0,0 +1,203 @@
+"""
+Common IO tests that apply to all grid formats.
+
+This module tests functionality that should work across all supported formats:
+- Basic read/write operations
+- Format conversions and round-trips
+- UGRID compliance
+- Common error handling
+"""
+
+import pytest
+import numpy as np
+import xarray as xr
+import uxarray as ux
+from pathlib import Path
+import tempfile
+import os
+from numpy.testing import assert_array_equal, assert_allclose
+from uxarray.constants import ERROR_TOLERANCE, INT_DTYPE, INT_FILL_VALUE
+
+try:
+    import constants
+except ImportError:
+    from . import constants
+
+current_path = Path(os.path.dirname(os.path.realpath(__file__)))
+
+# Define all testable format combinations
+# Format: (format_type, subpath, filename)
+IO_READ_TEST_FORMATS = [
+    ("ugrid", "ugrid/quad-hexagon", "grid.nc"),
+    ("ugrid", "ugrid/outCSne30", "outCSne30.ug"),
+    ("ugrid", "ugrid/outRLL1deg", "outRLL1deg.ug"),
+    ("mpas", "mpas/QU/480", "grid.nc"),
+    ("esmf", "esmf/ne30", "ne30pg3.grid.nc"),
+    ("exodus", "exodus/outCSne8", "outCSne8.g"),
+    ("exodus", "exodus/mixed", "mixed.exo"),
+    ("scrip", "scrip/outCSne8", "outCSne8.nc"),
+    ("icon", "icon/R02B04", "icon_grid_0010_R02B04_G.nc"),
+    ("fesom", "fesom/pi", None),  # Special case - multiple files
+]
+
+# Formats that support writing
+WRITABLE_FORMATS = ["ugrid", "exodus"]
+
+# Format conversion test pairs - removed for now as format conversion
+# requires more sophisticated handling than simple to_netcdf
+
+
+class TestIOCommon:
+    """Common IO tests across all formats."""
+
+    @pytest.mark.parametrize("format_name,subpath,filename", IO_READ_TEST_FORMATS)
+    def test_basic_read(self, format_name, subpath, filename):
```

Comment:
The fixture would act as a "basic read" test, since if any of the fixtures fail, then there is an issue in simply reading in a grid.